src/plugins/search/plugin.search.js

```diff
@@ -242,10 +250,12 @@ BookReader.prototype.cancelSearchRequest = function () {
   * @property {number} b
   * @property {number} t
   * @property {HTMLDivElement} [div]
+  * @property {number} matchIndex This is a fake field! not part of the API response. The index of the match that contains this box in total search results matches.
   */
 
 /**
  * @typedef {object} SearchInsideMatch
+ * @property {number} matchIndex This is a fake field! Not part of the API response. It is added by the JS.
```

Comment:
So I made the page numbers unique types because have a few types of "page numbers" floating around; but we'll probably only have one type of match number here, so feeling it was ok.